spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -112,38 +120,72 @@ public void launch(List<ModuleLaunchRequest> moduleLaunchRequests) {
 	 * {@literal --foo=bar} form.
 	 */
 	private String[] toArgArray(Map<String, String> args) {
-		String[] result = new String[args.size()];
-		int i = 0;
-		for (Map.Entry<String, String> kv : args.entrySet()) {
-			result[i++] = String.format("--%s=%s", kv.getKey(), kv.getValue());
+		if (args != null) {
+			String[] result = new String[args.size()];
+			int i = 0;
+			for (Map.Entry<String, String> kv : args.entrySet()) {
+				result[i++] = String.format("--%s=%s", kv.getKey(), kv.getValue());
+			}
+			return result;
+		}
+		else {
+			return new String[0];
 		}
-		return result;
 	}
 
-	public void launchAggregatedModules(List<ModuleLaunchRequest> moduleLaunchRequests, final String[] parentArgs) {
+	public void launchAggregatedModules(List<ModuleLaunchRequest> moduleLaunchRequests, Map<String,String> parentArgs) {
 		try {
 			List<String> mainClassNames = new ArrayList<>();
-			List<URL> jarURLs = new ArrayList<>();
+			LinkedHashSet<URL> jarURLs = new LinkedHashSet<>();
 			List<String> seenArchives = new ArrayList<>();
 			final List<String[]> arguments = new ArrayList<>();
-			// aggregate jars from all modules and extract their main Classes
-			for (ModuleLaunchRequest moduleLaunchRequest : moduleLaunchRequests) {
-				Resource resource = resolveModule(moduleLaunchRequest.getModule());
-				JarFileArchive jarFileArchive = new JarFileArchive(resource.getFile());
-				jarURLs.add(jarFileArchive.getUrl());
-				for (Archive archive : jarFileArchive.getNestedArchives(ArchiveMatchingEntryFilter.FILTER)) {
-					// avoid duplication based on unique JAR names
-					// TODO - read the metadata from the JARs, do proper version resolution on merge
-					String urlAsString = archive.getUrl().toString();
-					String jarNameWithExtension = urlAsString.substring(0, urlAsString.lastIndexOf("!/"));
-					String jarNameWithoutExtension = jarNameWithExtension.substring(jarNameWithExtension.lastIndexOf("/") + 1);
-					if (!seenArchives.contains(jarNameWithoutExtension)) {
-						seenArchives.add(jarNameWithoutExtension);
-						jarURLs.add(archive.getUrl());
+			if (!(parentArgs.containsKey(EXCLUDE_DEPENDENCIES_ARG) ||
+					parentArgs.containsKey(INCLUDE_DEPENDENCIES_ARG))) {
+				for (ModuleLaunchRequest moduleLaunchRequest : moduleLaunchRequests) {
+					Resource resource = resolveModule(moduleLaunchRequest.getModule());
+					JarFileArchive jarFileArchive = new JarFileArchive(resource.getFile());
+					jarURLs.add(jarFileArchive.getUrl());
+					for (Archive archive : jarFileArchive.getNestedArchives(ArchiveMatchingEntryFilter.FILTER)) {
+						// avoid duplication based on unique JAR names
```

Comment:
OTOH, it would be simply a matter of calling `resolve(null, modules, new Coordindates[0], new String[0])` tbh, so may be as well doing it ...